src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala

```diff
@@ -46,7 +46,11 @@ class ConsignmentService(
   }
 
   def updateTransferInitiated(consignmentId: UUID, userId: UUID): Future[Int] = {
-    consignmentRepository.updateTransferInitiated(consignmentId, userId, Timestamp.from(timeSource.now))
+    val updateTransferInitiatedStatus = consignmentRepository.updateTransferInitiated(consignmentId, userId, Timestamp.from(timeSource.now))
+    //Later, "Completed" might be changed to "InProgress"
+    val consignmentStatusRow = ConsignmentstatusRow(uuidSource.uuid, consignmentId, "Export", "Completed", Timestamp.from(timeSource.now))
+    consignmentStatusRepository.addConsignmentStatus(consignmentStatusRow)
+    updateTransferInitiatedStatus
   }
```

Comment:
Can you flatMap these two together. As it is, this will work because `Future` is eager but we could end up in a state where `updateTransferInitiated` fails but `addConsignmentStatus` succeeds. If you flatMap the two together, a bit like this:

```scala
for {
      updateTransferInitiatedStatus <- consignmentRepository.updateTransferInitiated(consignmentId, userId, Timestamp.from(timeSource.now))
      consignmentStatusRow = ConsignmentstatusRow(uuidSource.uuid, consignmentId, "Export", "Completed", Timestamp.from(timeSource.now))
      _ <- consignmentStatusRepository.addConsignmentStatus(consignmentStatusRow)
    } yield updateTransferInitiatedStatus
```
That should sort it.

Reply:
That's a very good point.